### ISSUE TITLE
Federated Learning User Story

### ISSUE THREAD

**Issue by jvmancuso** [OpenMined/PySyft-TensorFlow#42]
### Description
A user of PySyft will want to use TF in the way they would normally with PyTorch. Part of that means enabling federated learning as a use case. While we do not need to support yet all of the luxuries of the PyTorch side, we do want to demonstrate that the same use cases are solvable with TensorFlow.

This issue will be complete once a basic tutorial for federated learning has been implemented and completed.  This tutorial can be updated in a later issue/PR as Syft TF becomes more feature-complete (e.g. GradientTape has been implemented, etc.).

### Objectives/Key Results
- We have the demo code in a jupyter notebook
- We're training a federated model for multiple epochs
- Show loss decreasing
- Use PySyft sandbox for the demo

**Comment by karishnu** [comment]
Can the MNIST dataset be used to provide an image classification example?

**Comment by ProofofADA** [comment]
I would like to complete this issue.
